library/core/src/ptr/mod.rs

```diff
@@ -582,12 +582,13 @@ pub const fn invalid_mut<T>(addr: usize) -> *mut T {
 /// Convert an address back to a pointer, picking up a previously 'exposed' provenance.
 ///
 /// This is equivalent to `addr as *const T`. The provenance of the returned pointer is that of *any*
-/// pointer that was previously passed to [`expose_addr`][pointer::expose_addr] or a `ptr as usize`
-/// cast. If there is no previously 'exposed' provenance that justifies the way this pointer will be
-/// used, the program has undefined behavior. Note that there is no algorithm that decides which
-/// provenance will be used. You can think of this as "guessing" the right provenance, and the guess
-/// will be "maximally in your favor", in the sense that if there is any way to avoid undefined
-/// behavior, then that is the guess that will be taken.
+/// pointer that was previously exposed by passing it to [`expose_addr`][pointer::expose_addr],
+/// or a `ptr as usize` cast. In addition, memory which is outside of the control of Rust
+/// (MMIO registers, for example) is always considered to be exposed. If there is no 'exposed'
+/// provenance that justifies the way this pointer will be used, the program has undefined behavior.
```

Comment:
FWIW this relates to the other point you brought up recently, people thinking that they can circumvent the aliasing rules by using these functions. This sentence says quite clearly that that is not possible...

I was going to make a PR for that, but now it'd conflict with yours. I'll try making it a suggestions in here instead.